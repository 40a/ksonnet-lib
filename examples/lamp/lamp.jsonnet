local core = import "../../kube/core.libsonnet";
local kubeUtil = import "../../kube/util.libsonnet";

local container = core.v1.container;
local claim = core.v1.volume.claim;
local pod = core.v1.pod;
local deployment = kubeUtil.app.v1beta1.deployment;
local persistent = core.v1.volume.persistent;
local mount = core.v1.volume.mount;

{
  local lampPvcName = "my-lamp-site-data",
  local configStorageVolume =
    persistent.Default("site-data", lampPvcName),

  local mysqlContainer =
    container.Default("mysql", "mysql") +
    container.VolumeMounts([
      mount.FromVolume(configStorageVolume, "/var/lib/mysql") {
        subPath: "mysql"
      },
    ]),


  local phpContainer =
    container.Default("php", "php") +
    container.NamedPort("http", 80) +
    container.NamedPort("https", 443) +
    container.VolumeMounts([
      mount.FromVolume(configStorageVolume, "/var/www/html") {
        subPath: "html"
      },
    ]),

  "lamp-pod.json": kubeUtil.app.v1.pod.FromContainer(mysqlContainer) {
      spec+: {
        containers+: [phpContainer],
      } +
      pod.spec.Volumes([configStorageVolume]),
    },

  // NOTE: This is a total guess.
  "lamp-pvc.json": claim.DefaultPersistent(
    "lamp-test", lampPvcName, ["ReadWriteMany"], "20Gi"),
}
