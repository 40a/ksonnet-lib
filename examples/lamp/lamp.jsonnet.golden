{
   "lamp-pod.json": {
      "apiVersion": "v1",
      "kind": "Pod",
      "metadata": {
         "labels": {
            "app": "mysql"
         }
      },
      "spec": {
         "containers": [
            {
               "env": [ ],
               "image": "mysql",
               "imagePullPolicy": "Always",
               "name": "mysql",
               "ports": [ ],
               "volumeMounts": [
                  {
                     "mountPath": "/var/lib/mysql",
                     "name": "site-data",
                     "readOnly": false,
                     "subPath": "mysql"
                  }
               ]
            },
            {
               "env": [ ],
               "image": "php",
               "imagePullPolicy": "Always",
               "name": "php",
               "ports": [
                  {
                     "containerPort": 80,
                     "name": "http"
                  },
                  {
                     "containerPort": 443,
                     "name": "https"
                  }
               ],
               "volumeMounts": [
                  {
                     "mountPath": "/var/www/html",
                     "name": "site-data",
                     "readOnly": false,
                     "subPath": "html"
                  }
               ]
            }
         ],
         "volumes": [
            {
               "name": "site-data",
               "persistentVolumeClaim": {
                  "claimName": "my-lamp-site-data"
               }
            }
         ]
      }
   },
   "lamp-pvc.json": {
      "apiVersion": "v1",
      "kind": "PersistentVolumeClaim",
      "metadata": {
         "annotations": {
            "volume.beta.kubernetes.io/storage-class": "fast"
         },
         "name": "my-lamp-site-data",
         "namespace": "lamp-test"
      },
      "spec": {
         "accessModes": [
            "ReadWriteMany"
         ],
         "resources": {
            "requests": {
               "storage": "20Gi"
            }
         }
      }
   }
}
